### ISSUE TITLE
Gradient explosion during training

### ISSUE THREAD

**Issue by zhan8855** [ArcInstitute/evo2#105]
Thank you for your awesome work!

I am trying to launch evo2 training, but I find that the gradient of HCL layer will explode after only a few training steps. The problem might come from [https://github.com/Zymrael/vortex/blob/f243e8ec5da8374be082a77056c0a447e7fa9231/vortex/model/model.py#L392](https://github.com/Zymrael/vortex/blob/f243e8ec5da8374be082a77056c0a447e7fa9231/vortex/model/model.py#L392). I would greatly appreciate it if you could share how did you address this problem during Evo2 training.

Thank you very much in advance.

**Comment by jstjohn** [comment]
Training was done with a different formulation of the forward pass in a different repo. Look for links to savanna. Alternatively you could try the bionemo codebase that also has the savanna style training functionality with tutorial notebooks covering data conversion from fasta etc. Sent from my iPhoneOn Apr 17, 2025, at 9:14 AM, zhan8855 ***@***.***> wrote:﻿
  Thank you for your awesome work!
I am trying to launch evo2 training, but I find that the gradient of HCL layer will explode after only a few training steps. The problem might come from https://github.com/Zymrael/vortex/blob/f243e8ec5da8374be082a77056c0a447e7fa9231/vortex/model/model.py#L392. I would greatly appreciate it if you could share how did you address this problem during Evo2 training.
Thank you very much in advance.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



zhan8855 created an issue (ArcInstitute/evo2#105)
Thank you for your awesome work!
I am trying to launch evo2 training, but I find that the gradient of HCL layer will explode after only a few training steps. The problem might come from https://github.com/Zymrael/vortex/blob/f243e8ec5da8374be082a77056c0a447e7fa9231/vortex/model/model.py#L392. I would greatly appreciate it if you could share how did you address this problem during Evo2 training.
Thank you very much in advance.

**Comment by garykbrixi** [comment]
Thanks, I am marking this as resolved
As John mentioned above, vortex is for inference only. Zhan you should take a look at [bionemo](https://docs.nvidia.com/bionemo-framework/2.5/user-guide/examples/bionemo-evo2/fine-tuning-tutorial/) or [savanna](https://github.com/Zymrael/savanna) for finetuning.
